### DISCUSSION TITLE
Can i change how many mutations are created?

### DISCUSSION THREAD

**Discussion by C0DK** [stryker-mutator/stryker-net#2086] [Q&A] [answered]
I see the number of mutations change slightly, presumably as a function of the code length. Is it actually doing all the possible mutations it can, or is it some function of them? Can I edit this? As I currently only run Stryker manually rarely to validate the quality of my code, I'd love to be able to make it run for longer but hopefully just catch more cases, if possible. I cannot see if there is some element of randomness.

**Comment by rouke-broersma** [comment, marked as answer]
We always generate all possible mutations. There should not be any element of randomness. If you experience an element of randomness on the **total** amount of mutants on the exact same codebase with the exact same stryker version then we have a bug. If you experience an element of randomness in the mutants per status then we might have a bug on for example coverage analysis or your tests might not be entirely deterministic. Or if you experience **new** mutations when you add more code then that is to be expected as there would be more code to mutate. New versions of stryker could also influence the amount of mutants due to adding new mutators or fixing bugs.

**Reply by C0DK** [reply]
I might have simply changed the codebase significantly - the numbers just seemed to change a lot. I'll keep my eyes open, thanks! :) 

(I am using autofixture, so they are not entirely deterministic, but hopefully, I've written sufficiently nice tests that the same areas will be cover :P )
